prboom2/src/SDL/i_system.c

```diff
@@ -318,6 +328,32 @@ const char *I_DoomExeDir(void)
   return base;
 }
 
+const char *I_ExeDir(void)
+{
+  extern char **dsda_argv;
+
+  static const char current_dir_dummy[] = {"."}; // proff - rem extra slash 8/21/03
+  static char *base;
+  if (!base)        // cache multiple requests
+    {
+      size_t len = strlen(*dsda_argv);
+      char *p = (base = (char*)Z_Malloc(len+1)) + len - 1;
+      strcpy(base,*dsda_argv);
+      while (p > base && *p!='/' && *p!='\\')
+        *p--=0;
+      if (*p=='/' || *p=='\\')
+        *p--=0;
+      if (strlen(base) < 2 || !M_WriteAccess(base))
+      {
+        Z_Free(base);
+        base = (char*)Z_Malloc(1024);
+        if (!M_getcwd(base, 1024) || !M_WriteAccess(base))
+          strcpy(base, current_dir_dummy);
```

Comment:
Can we just do `strcpy(base, ".");`? Does this dummy variable need to persist for some reason?

Reply:
No there is no reason it needs to persist. I simply copied over the code from the Windows function, didnt even try to understand it.
Made a new commit.